frontend/src/views/WheresMyGene/components/RightSideBar/index.tsx

```diff
@@ -0,0 +1,77 @@
+import { ButtonIcon } from "czifui";
+import { Children, memo, ReactElement, ReactNode } from "react";
+import {
+  Position,
+  SideBar as SideBarWrapper,
+} from "src/components/common/SideBar/style";
+import { CELL_INFO_SIDEBAR_WIDTH_PX } from "../CellInfoSideBar/style";
+import { RightSideBarPositioner, StyledTitle, HeaderContainer } from "./style";
+
+export interface RightSidebarProperties {
+  handleClose: () => void;
+  title: string;
+}
+interface Props {
+  children: ReactNode;
+  width?: number;
+  testId?: string;
+}
+
+// Values are in CSS vh units
+const FULL_MAX_HEIGHT = 100;
+const UPPER_SECTION_MAX_HEIGHT = 60;
+const DRAWER_MAX_HEIGHT = FULL_MAX_HEIGHT - UPPER_SECTION_MAX_HEIGHT;
+
+export default memo(function RightSideBar({
+  children,
+  width = CELL_INFO_SIDEBAR_WIDTH_PX,
+  testId,
+}: Props): JSX.Element | null {
+  const content = Children.map(
+    children,
+    (child) => child as ReactElement<RightSidebarProperties>
+  );
+
+  if (!content) return null;
+
+  const isSplit = content.length === 2;
```

Comment:
we determine if the right sidebar should be split if there are two children components